### PR TITLE
Add non indicator stix objects and add Relationships between Indicators

### PR DESCRIPTION
**Pull request by MosheEichler**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/28814

## Description
Add description to stix indicators, add relationships between 2 Indicators

## Minimum version of Cortex XSOAR
- [x] 5.5.0
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [x] Yes
       - We changed the `share_level` key in the context to `TLP`
   - [ ] No

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/FeedTAXII/ReleaseNotes/1_0_13.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+
+#### Integrations
+##### TAXII Feed
+- Added new fields to the indicator
```
Comment (amshamah419) [inline Packs/FeedTAXII/ReleaseNotes/1_0_13.md:4]:
```suggestion
- Added new fields to the indicator.
```

File: Packs/FeedTAXII/ReleaseNotes/1_0_13.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+- Added relationships between indicators
```
Comment (amshamah419) [inline Packs/FeedTAXII/ReleaseNotes/1_0_13.md:5]:
```suggestion
- Added support for relationships between indicators.
```

File: Packs/FeedTAXII/ReleaseNotes/1_0_13.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+#### Indicator Fields
+##### New: STIX Indicator Description
+- The stix indicator description
```
Comment (amshamah419) [inline Packs/FeedTAXII/ReleaseNotes/1_0_13.md:9]:
```suggestion
- Description for the STIX indicator.
```

File: Packs/FeedTAXII/ReleaseNotes/1_0_13.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+##### New: STIX Indicator Name
+- The stix indicator name
```
Comment (amshamah419) [inline Packs/FeedTAXII/ReleaseNotes/1_0_13.md:12]:
```suggestion
- Name for the STIX indicator.
```

File: Packs/FeedTAXII/Integrations/FeedTAXII/FeedTAXII.py [hunk 5]
```diff
@@ -806,6 +853,25 @@ def _poll_collection(self, poll_service, begin, end):
                 stream=True
             )
 
+        for observable in observables:
+
+            indicator_ref = observable.get('indicator_ref')
+            if indicator_ref:
```
Comment (reutshal) [inline Packs/FeedTAXII/Integrations/FeedTAXII/FeedTAXII.py:859]:
```suggestion
            if indicator_ref := observable.get('indicator_ref')
```

File: Packs/FeedTAXII/Integrations/FeedTAXII/FeedTAXII.py [hunk 6]
```diff
@@ -918,7 +984,11 @@ def package_extract_properties(package):
 
 def observable_extract_properties(observable):
     """Extracts properties from observable"""
-    result = {}
+    result: Dict[str, str] = {}
+
+    id_ref = observable.get('id')
+    if id_ref:
```
Comment (reutshal) [inline Packs/FeedTAXII/Integrations/FeedTAXII/FeedTAXII.py:990]:
```suggestion
    if id_ref := observable.get('id')
```

File: Packs/FeedTAXII/Integrations/FeedTAXII/FeedTAXII.py [hunk 7]
```diff
@@ -933,6 +1003,82 @@ def observable_extract_properties(observable):
...
+
+    confidence = next((c for c in indicator if c.name == 'Confidence'), None)
+    if confidence is not None:
+        value = next((c for c in confidence if c.name == 'Value'), None)
+        if value is not None:
+            value = value.text
+            result['confidence'] = value
+
+    indicated_ttp = indicator.find_all('Indicated_TTP')
+    if indicated_ttp:
```
Comment (reutshal) [inline Packs/FeedTAXII/Integrations/FeedTAXII/FeedTAXII.py:1028]: Same here

File: Packs/FeedTAXII/Integrations/FeedTAXII/FeedTAXII.py [hunk 8]
```diff
@@ -933,6 +1003,82 @@ def observable_extract_properties(observable):
     return result
 
 
+def indicator_extract_properties(indicator) -> Dict[str, Any]:
+    """Extracts properties from indicator"""
```
Comment (reutshal) [inline Packs/FeedTAXII/Integrations/FeedTAXII/FeedTAXII.py:1007]: If you added a docstring please do it properly - with args and returns

File: Packs/FeedTAXII/Integrations/FeedTAXII/FeedTAXII.py [hunk 9]
```diff
@@ -933,6 +1003,82 @@ def observable_extract_properties(observable):
...
+        for ttp_value in indicated_ttp:
+            ttp = next((c for c in ttp_value if c.name == 'TTP'), None)
+            if ttp is not None:
+                value = ttp.get('idref')
+                result['ttp_ref'].append(value)
+
+    return result
+
+
+def ttp_extract_properties(ttp, behavior):
```
Comment (reutshal) [inline Packs/FeedTAXII/Integrations/FeedTAXII/FeedTAXII.py:1039]: Please add a short docstring and add the returned value typing to the method's signature

File: Packs/FeedTAXII/Integrations/FeedTAXII/FeedTAXII.py [hunk 10]
```diff
@@ -975,25 +1143,57 @@ def test_module(client, *_):
...
             if client.tlp_color:
                 indicator_obj['fields']['trafficlightprotocol'] = client.tlp_color
 
             indicators.append(indicator_obj)
 
+    # Create the indicators from the ttps
+    ttps = client.ttps
+    for item in ttps.values():
+        indicator = item.get('indicator')
+        if indicator:
```
Comment (reutshal) [inline Packs/FeedTAXII/Integrations/FeedTAXII/FeedTAXII.py:1178]:
```suggestion
        if indicator := item.get('indicator')
```

File: Packs/FeedTAXII/Integrations/FeedTAXII/FeedTAXII_test.py [hunk 11]
```diff
@@ -98,6 +98,16 @@ def test_fetch_indicators(self, mocker):
                 assert res == expected
 
 
+def test_poll_collection():
```
Comment (reutshal) [inline Packs/FeedTAXII/Integrations/FeedTAXII/FeedTAXII_test.py:101]: Please add "Given When Then" to the test, what exactly are you testing here?

File: Packs/FeedTAXII/Integrations/FeedTAXII/FeedTAXII_test.py [hunk 12]
```diff
@@ -98,6 +98,16 @@ def test_fetch_indicators(self, mocker):
...
+    import requests_mock
+    from FeedTAXII import TAXIIClient
+    client = TAXIIClient(collection='a collection')
+    with requests_mock.Mocker() as m:
+        m.post('google.com')
+        res = client._poll_collection('http://google.com', '1', '2')
+    assert res
+
+
```
Comment (reutshal) [inline Packs/FeedTAXII/Integrations/FeedTAXII/FeedTAXII_test.py:110]: It looks like you are almost not testing the code you added.
Please go over it and make sure to add relevant tests.

File: Packs/FeedTAXII/Integrations/FeedTAXII/FeedTAXII_test/TestCommands/indicators_results.json [hunk 13]
```diff
@@ -749,4 +849,4 @@
             "value": "http://111.90.149.220/banks/Desjardins/a40b89886fe5eb23fc134a53e1bd0af2"
         }
     }
-]
+]
```
Comment (reutshal) [inline Packs/FeedTAXII/Integrations/FeedTAXII/FeedTAXII_test/TestCommands/indicators_results.json:852]: Please make sure there are no secrets in this file

**Review by reutshal — CHANGES_REQUESTED** [review]
Nice job! Please see my commants.
Dont forget to add the new outputs to the yml file.

File: Packs/FeedTAXII/Integrations/FeedTAXII/FeedTAXII.yml [hunk 14]
```diff
@@ -159,6 +159,33 @@ script:
...
       description: The indicator type.
       type: String
+    - contextPath: TAXII.Indicator.title
+      description: The indicator title.
+      type: String
+    - contextPath: TAXII.Indicator.description
+      description: The indicator description.
+      type: String
+    - contextPath: TAXII.Indicator.stixindicatordescription
+      description: The indicator description.
```
Comment (reutshal) [inline Packs/FeedTAXII/Integrations/FeedTAXII/FeedTAXII.yml:169]: You have the same descriptions here, same in the "title" output.
Please be more specific so customers will understand the difference.

File: Packs/FeedTAXII/Integrations/FeedTAXII/FeedTAXII_test/TestCommands/collection_example.xml [hunk 15]
```diff
@@ -0,0 +1,284 @@
...
+                                <URIObj:Value condition="Equals">http://example.org/</URIObj:Value>
+                            </cybox:Properties>
+                        </cybox:Object>
+                    </cybox:Observable>
+                </stix:Observables>
+            </stix:STIX_Package>
+        </taxii_11:Content>
+        <taxii_11:Timestamp_Label>2021-10-20T06:43:08.853465+00:00</taxii_11:Timestamp_Label>
+    </taxii_11:Content_Block>
+</taxii_11:Poll_Response>
```
Comment (reutshal) [inline Packs/FeedTAXII/Integrations/FeedTAXII/FeedTAXII_test/TestCommands/collection_example.xml:284]: Please make sure there aren't secrets in this file

File: Packs/FeedTAXII/Integrations/FeedTAXII/FeedTAXII_test/TestCommands/indicators_example.json [hunk 16]
```diff
@@ -0,0 +1,167 @@
...
+        "fields": {},
+        "rawJSON": {
+            "stix_ttp_title": "Email Emmbedded URL",
+            "ttp_description": "Target Users via Email by adding a malicious URL",
+            "type": "Attack Pattern",
+            "indicator": "URL embedded in Email",
+            "value": "URL embedded in Email"
+        }
+    }
+]
```
Comment (reutshal) [inline Packs/FeedTAXII/Integrations/FeedTAXII/FeedTAXII_test/TestCommands/indicators_example.json:167]: Same here - check for secrets

File: Packs/FeedTAXII/Integrations/FeedTAXII/README.md [hunk 17]
```diff
@@ -1,166 +1,308 @@
...
...r: http://example.org/mellenium/index.php<br>type: URL<br>indicator_ref: opensource:Observable-810ca3be-d3b7-460f-ae17-a73bcd899633<br>stix_title: URL: http://example.org/mellenium/index.php...<br>stix_description: URL: http://example.org/mellenium/index.php\| isOnline:yes\| dateVerified:2021-10-19T04:13:43+00:00<br>TLP: WHITE<br>stix_indicator_name: phishTank.com id:7324342 with malicious URL:http://example.org/melle...<br>stix_indicator_description: This URL:[http://example.org/mellenium/index.php] was identified by phishtank.com as part of a phishing email which appears to be targeting Bank Millennium This URL appears to still be online as of 2021-10-19T04:13:43+00:00. More detailed infomation can be found at http://www.phishtank.com/phish_detail.php?phish_id=7324342<br>confidence: High<br>ttp_ref: opensource:ttp-86c046db-2a49-4a90-8abc-3e274b11027d,<br>opensource:ttp-c819f3ef-fbc3-4077-8d56-bf619c8d9b29<br>relationships: {'stix_ttp_title': 'Email Emmbedded URL', 'ttp_description': 'Target Users via Email by adding a malicious URL', 'type': 'Attack Pattern', 'indicator': 'URL embedded in Email', 'value': 'URL embedded in Email'}<br>value: http://example.org/mellenium/index.php |
```
Comment (reutshal) [inline Packs/FeedTAXII/Integrations/FeedTAXII/README.md:308]: Please fix readme as discussed

**Review by reutshal — CHANGES_REQUESTED** [review]
Please see my final comments

File: Packs/FeedTAXII/Integrations/FeedTAXII/FeedTAXII_test/TestCommands/collection_example.xml (hunk 15 above)
Comment (MosheEichler) [reply Packs/FeedTAXII/Integrations/FeedTAXII/FeedTAXII_test/TestCommands/collection_example.xml:284]: Screen Shot 2021-11-07 at 12 42 41

File: Packs/FeedTAXII/Integrations/FeedTAXII/FeedTAXII_test/TestCommands/indicators_example.json (hunk 16 above)
Comment (MosheEichler) [reply Packs/FeedTAXII/Integrations/FeedTAXII/FeedTAXII_test/TestCommands/indicators_example.json:167]: Screen Shot 2021-11-07 at 12 42 51

File: Packs/FeedTAXII/Integrations/FeedTAXII/FeedTAXII.yml [hunk 18]
```diff
@@ -159,6 +159,33 @@ script:
...
+      description: The indicator title.
+      type: String
+    - contextPath: TAXII.Indicator.description
+      description: The observable description.
+      type: String
+    - contextPath: TAXII.Indicator.stixindicatordescription
+      description: The indicator description.
+      type: String
+    - contextPath: TAXII.Indicator.stixindicatorname
+      description: The indicator title.
```
Comment (reutshal) [inline Packs/FeedTAXII/Integrations/FeedTAXII/FeedTAXII.yml:172]: What about the title description? Please make sure you don't have the same descriptions to different paths

**Review by reutshal — APPROVED** [review]

**Pull request merged**